docs/src/references/c8y-configuration-management.md

```diff
@@ -264,4 +271,75 @@ When there is no coresponding config file on the thin-edge device the plugin sen
+    * `path` is chosen by the external-device; it may contain folders (e.g. `foo/bar/config-file.conf`)
+    * on an incoming PUT request `tedge_agent` stores the contained file to `/http-root/<path>`
+    * on an incoming GET request `tedge_agent` sends the file `/http-root/<path>` to the requester (i.E. the external device)
+    * `tedge_agent` does not force any predefined folder structure within `/http-root/`. However for configuration management it is best practice to use folder structure as below:
+      * `/http-root/config/current/<child-id>/<config filename>`<br/>
+         for config-files provided by the external device to the thin-edge device (current state - or latest known state)<br/>
+      * `/http-root/config/desired/<child-id>/<config filename>`<br/>
+        for config-files to be consumed by the external device (desired state)
+  * `tedge_agent` notifies over MQTT any change of a file under `/http-root/` (e.g. when uploaded by a external-device or updated by a local process). The notification messages are published on the topic `tedge/filetransfer_change/{path}`.<br/>
+    TO-BE-DEFINED: Content of payload?
+      
+      Example:
+      * HTTP PUT URL: `http://<thin-edge IP address>/tedge/filetransfer/config/current/child1/bar.conf`
+      * relating MQTT notification topic: `/tedge/filetransfer_change/config/current/child1/bar.conf`
+
+  * `c8y_configuration_plugin` allows to handle two versions for each configuration file (_current_ and _desired_).
+    * The configuration file's record `path` is treated as path of the _current_ version of a configuration file (see to section 'Configuration' above).
+    * The configuration file's record `desired` can be used to define the path of the _desired_ version of a configuration file. If `desired` is not defined, then `path` is treated as _desired_ version also.
+    * On c8y read request for a configuration, `c8y_configuration_plugin` returns the `current` file content.
+    * On c8y write request for a configuration, `c8y_configuration_plugin` updates the `desired` file content and waits for the `current` content to be updated to return the operation status to cumulocity.  
+    * The current behavior of c8y_configuration_plugin can be simulated with desired = current = path when only a path is provided.
+
+Example Plugin Config:
+```shell
+$ cat /etc/tedge/c8y/c8y-configuration-plugin.toml
+files = [
+    { type = 'bar.conf', child_id = 'child1', current = '/http-root/config/current/child1/bar.conf', desired = '/http-root/config/desired/child1/bar.conf' } }
+  ]
+```
+
+Example Flow:
+
+**Start Behaviour:**
+  * external device child1: starts
+  * external device child1: sends all its config files with HTTP PUT requests to the thin-edge device; 
+
+    Example: HTTP PUT to `http://<thin-edge IP address>/tedge/filetransfer/current/child1/bar.conf`
+
+**Device-to-Cloud Behaviour:**
```

Comment:
When and how are declared the capabilities of the child devices?

Reply:
Details are added now, see section [Declaration of Capabilities for Child-Devices](https://github.com/thin-edge/thin-edge.io/blob/a6af33a1aaa3dbc8b8cfec94c1d81d093c09ff7f/docs/src/references/c8y-configuration-management.md#declaration-of-capabilities-for-child-devices)